### PR TITLE
Integrate historic data

### PR DESCRIPTION
**Pull request by CBarrett2**
Added an option for basic historic data, no period/product selectors atm but they would be simple to add afterwards. This also makes it pretty simple to add in support for a live stream. There is a button to switch between 'live' and generated data, and a class for the historic feed which adds the ability to cancel a request that's been made (invalidateCallback)

### CODE REVIEW

**Comment by CBarrett2** [comment]
Resolves #83

File: src/assets/js/data/feed/coinbase/historicFeed.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+(function(sc) {
+    'use strict';
+
+    sc.data.feed.coinbase.historicFeed = function() {
```
Comment (tlsim) [inline src/assets/js/data/feed/coinbase/historicFeed.js]: Not sure we need this extra class. Is `invalidateCallback` intended to stop a slow to return coinbase callback rendering if the user switches back to generated data? Could main.js `historicCallback` check a boolean instead?

Comment (CBarrett2) [inline src/assets/js/data/feed/coinbase/historicFeed.js]: Yes, that's right :) Possibly, although I don't think that's quite as sturdy - what if someone tries to load live data, changes back to generated, then back to live again, and for some reason the first call comes back after the second? Unlikely, perhaps. We could probably move all this to main.js for now if you like?

Comment (CBarrett2) [inline src/assets/js/data/feed/coinbase/historicFeed.js]: Also a boolean value doesn't cover if you want to potentially make multiple historic requests - as in when we get to changing period/instrument - we'll want the option to change quickly between periods/instruments etc, so I think the integer is better.

File: src/assets/js/main.js [hunk 2]
```diff
@@ -121,9 +121,37 @@
...
+    d3.select('#type-selection')
+        .on('change', function() {
+            var type = d3.select(this).property('value');
+            if (type === 'live') {
+                historicFeed(historicCallback);
+            } else if (type === 'generated') {
+                historicFeed.invalidateCallback();
+                dataModel.data = fc.data.random.financial()(250);
+                resetToLive();
+                render();
```
Comment (CBarrett2) [inline src/assets/js/main.js]: Pull out as method

**Pull request closed without merging**